download_plugins.py

```diff
@@ -1,84 +1,124 @@
+def download_plugin(plugin, version=None):
     download_url = ""
-    
+
     if version is None:
-        download_url = plugin_base_url + "/latest/%s.hpi" % plugin
+        download_url = (plugin_base_url +
+                        "/latest/{}.hpi".format(plugin))
     else:
-        download_url = plugin_base_url + "/download/plugins/%(plugin)s/%(version)s/%(plugin)s.hpi" % {'plugin': plugin, 'version': version}
+        download_url = (plugin_base_url +
+                        "/download/plugins/" +
+                        "{plugin}/{version}/{plugin}.hpi".format(
+                            plugin=plugin, version=version
+                        ))
 
     plugin_download = requests.get(download_url, stream=True)
 
-    if plugin_download.status_code is 200:
+    if plugin_download.status_code == 200:
         destination_path = download_directory + "/" + plugin + ".hpi"
 
-        with open(destination_path, 'wb') as data:
+        with open(destination_path, "wb") as data:
             for chunk in plugin_download.iter_content(chunk_size=128):
                 data.write(chunk)
 
-        print "Downloaded", plugin
-        
+        print("Downloaded", plugin)
+
     else:
-        print "Error downloading %s. Response:" % plugin
-        print plugin_download.text
+        print("Error downloading {}. Response:".format(plugin))
+        print(plugin_download.text)
+        global exit_code
         exit_code = 1
 
-    installed_plugins.append(plugin)
 
 # Install each plugin in the supplied json file along with dependencies.
 def install_plugins():
     for plugin, version in plugins.items():
-        print "**** Install Plugin: %s ****" % plugin
+        print("\n**** Install Plugin: {} *****".format(plugin))
         download_plugin(plugin, version)
         get_dependencies(plugin)
 
+    print("\nDownloading all dependencies")
+    print("*****************************************")
+    for plugin, version in stored_plugins.items():
+        download_plugin(plugin, version)
+
+    if multi_version_plugins:
+        for plugin in multi_version_plugins:
+            print(dedent("""\
+            Warning: Multiple versions of {plugin} found in dependencies.
+                Version {version} downloaded\
+            """.format(plugin=plugin, version=stored_plugins[plugin])))
+            # print("Warning: Multiple versions of {} " +
+            #       "found in dependencies.".format(plugin))
+            # print("Version {} downloaded".format(stored_plugins[plugin]))
+
 plugins_file_path = str(sys.argv[1])
 download_directory = str(sys.argv[2])
 
 try:
-    with open(plugins_file_path, 'r') as file:
-        plugins = yaml.load(file)['plugins']
+    with open(plugins_file_path, "r") as file:
+        plugins = yaml.load(file)["plugins"]
 except (IOError, ValueError):
-    print "Unable to load plugin yaml file"
+    print("Unable to load plugin yaml file")
     sys.exit(1)
```

Comment:
You can [combine these two lines](https://docs.python.org/2/library/sys.html#sys.exit) like `sys.exit('Hey, I'm the error message!')`